src/store/socket.js

```diff
@@ -251,6 +255,7 @@ class LiveSession {
       id: player.id,
       isDead: player.isDead,
       isVoteless: player.isVoteless,
+      isOnBlock: player.isOnBlock,
```

Comment:
Remnant of an older version? It's not a player attribute. (anymore)

Reply:
I'm lost by this - it is a player attribute!

Reply:
Ah, my bad, based on how you added the logic to the Player.vue (with an emitted event and handled in the TownSquare.vue) I assumed you implemented the logic like a nomination. There can only ever be one player "on the block", so I don't think it is a good approach to add this as a player property. Please move it to the session module and treat it similarly to the nomination, but with only a single integer instead of two.
`session.marked = -1` would be the default (nobody is marked) and when someone is marked at the end of a vote, it would be set to the index of that player's seat. This would probably also fix the bug I discovered with the player element not updating due to an unwatched property being changed and it would also remove the need to have a `setOnBlock` mutator that touches every single player.

Reply:
I did originally code it like that, but I simply can't see how to make it not break when players get removed/added whilst someone is on the block. (The nomination + voting stuff is also broken if they are active when players change, as per https://github.com/bra1n/townsquare/issues/164.)

Reply:
I'm working on a fix for swapping / removing / moving players during a nomination, it's actually quite simple. That would also be the place where you add a hook to remove or update the mark if the player is (re)moved. It's 2 lines in one place, I'll link it here when I'm done. :-)

Reply:
@nicfreeman1209 there's a fix for this here: https://github.com/bra1n/townsquare/pull/165/commits/71234f3ef90793a743213f856e72abab89b1ce2a#diff-17a6801e9bf7c77c2b6aa62f71970a59597c7b9b866893aec58c830b24261321

Reply:
Thanks! It's easiest for me to wait until that commit is on your main branch before doing more here.

(Annoyingly, it gives a merge conflict inside the changelog & who knows what sort of tantrum gut might throw if I fork the fixed branch, resolve the conflict with a commit, PR that into this branch and then try to PR this branch back into your main one...)